tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -1749,6 +1749,91 @@ HRESULT DxcCompilerAdapter::WrapCompile(
 }
 //////////////////////////////////////////////////////////////
 
+namespace
+{
+class DxcCompilerAdapterInterface : public IDxcCompilerAdapter, public IDxcCompiler3
+{
+private:
+  DXC_MICROCOM_TM_REF_FIELDS()
+  IDxcCompiler3 *m_pDxcCompiler3;
+  DxcCompilerAdapter m_DxcCompilerAdapter;
+
+public:
+  DxcCompilerAdapterInterface(IMalloc *pMalloc)
+      : m_dwRef(0)
+      , m_pMalloc(pMalloc)
+      , m_pDxcCompiler3(nullptr)
+      , m_DxcCompilerAdapter(this, pMalloc) 
+  {
+  }
+  
+  DXC_MICROCOM_TM_ALLOC(DxcCompilerAdapterInterface)
+  
+  HRESULT STDMETHODCALLTYPE QueryInterface(REFIID iid, void **ppvObject) override {
+    HRESULT hr = DoBasicQueryInterface<IDxcCompilerAdapter, IDxcCompiler3>(this, iid, ppvObject);
+    if (FAILED(hr)) {
+        return DoBasicQueryInterface<IDxcCompiler, IDxcCompiler2>(&m_DxcCompilerAdapter, iid, ppvObject);
```

Comment:
I suppose this may be where the mismatching SetDxcCompiler call might cause a problem?